src/shared/tools/xml_parser.h

```diff
@@ -0,0 +1,387 @@
+/* ------------------------------------------------------------------------- *
+ *                                SPHinXsys                                  *
+ * ------------------------------------------------------------------------- *
+ * SPHinXsys (pronunciation: s'finksis) is an acronym from Smoothed Particle *
+ * Hydrodynamics for industrial compleX systems. It provides C++ APIs for    *
+ * physical accurate simulation and aims to model coupled industrial dynamic *
+ * systems including fluid, solid, multi-body dynamics and beyond with SPH   *
+ * (smoothed particle hydrodynamics), a meshless computational method using  *
+ * particle discretization.                                                  *
+ *                                                                           *
+ * SPHinXsys is partially funded by German Research Foundation               *
+ * (Deutsche Forschungsgemeinschaft) DFG HU1527/6-1, HU1527/10-1,            *
+ *  HU1527/12-1 and HU1527/12-4.                                             *
+ *                                                                           *
+ * Portions copyright (c) 2017-2023 Technical University of Munich and       *
+ * the authors' affiliations.                                                *
+ *                                                                           *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may   *
+ * not use this file except in compliance with the License. You may obtain a *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.        *
+ *                                                                           *
+ * ------------------------------------------------------------------------- */
+/**
+ * @file    xml_parser.h
+ * @brief   XML parser based on tinyxml2 (https://github.com/leethomason/tinyxml2)
+ * @author	Chi Zhang 
+ */
+#pragma once
+
+#include "tinyxml2.h"
+
+#include "base_data_package.h"
+#include "sph_data_containers.h"
+
+#include <iostream>
+#include <string>
+#include <cstdio>
+#include <cassert>
+#include <charconv>
+
+#include <fstream>
+#include <filesystem>
+
+namespace fs = std::filesystem;
+#define assertm(exp, msg) assert(((void)msg, exp))
+
+
+namespace SPH
+{	
+	// Eigen::IOFormat CleanFmt(Eigen::StreamPrecision, 0, ", ", "\n", "[", "]");
+	// Eigen::IOFormat CommaInitFmt(Eigen::StreamPrecision, Eigen::DontAlignCols, ", ", ", ", "", "", " << ", ";");
+	// Eigen::IOFormat CleanFmt(Eigen::StreamPrecision, 0, ", ", "\n", "[", "]");
+	// Eigen::IOFormat OctaveFmt(Eigen::StreamPrecision, 0, ", ", ";\n", "", "", "[", "]");
+	// Eigen::IOFormat HeavyFmt(Eigen::FullPrecision, 0, ", ", ";\n", "[", "]", "[", "]");
+
+	template < typename DataType >
+	inline std::string DataToString( const DataType &value )
+	{
+		std::ostringstream out;
+    	out.precision(15);
+    	out << std::fixed << value;
+    	return std::move(out).str();
+		//return std::to_string(value);
```

Comment:
Another option : 

template<typename T> 
std::string data_to_string(const T& value)
{
    std::array<char, 15> str;

    auto res = std::to_chars(str.data(), str.data() + str.size(), value);
    return std::string(str.data(), res.ptr);
}

Seems not work due to the compiler version.